control-plane/pkg/reconciler/consumer/consumer.go

```diff
@@ -0,0 +1,422 @@
+	if c.Spec.Reply.URLReply != nil {
+		destination, err := r.Resolver.URIFromDestinationV1(ctx, c.Spec.Reply.URLReply.Destination, c)
+		if err != nil {
+			return fmt.Errorf("failed to resolve reply destination: %w", err)
+		}
+		egress.ReplyStrategy = &contract.Egress_ReplyUrl{
+			ReplyUrl: destination.String(),
+		}
+		return nil
+	}
+	if c.Spec.Reply.TopicReply != nil {
+		egress.ReplyStrategy = &contract.Egress_ReplyToOriginalTopic{}
+		return nil
+	}
+
+	return nil
+}
+
+func reconcileFilters(c *kafkainternals.Consumer) *contract.Filter {
+	if c.Spec.Filters == nil {
+		return nil
+	}
+	if c.Spec.Filters.Filter != nil {
+		return &contract.Filter{Attributes: c.Spec.Filters.Filter.Attributes}
+	}
+
+	return nil
+}
+
+type contractMutatorFunc func(logger *zap.Logger, ct *contract.Contract, c *kafkainternals.Consumer) int
+
+func addResource(resource *contract.Resource) contractMutatorFunc {
+	return func(logger *zap.Logger, ct *contract.Contract, c *kafkainternals.Consumer) int {
+		return coreconfig.AddOrUpdateResourceConfig(ct, resource, coreconfig.FindResource(ct, c.GetUID()), logger)
+	}
+}
+
+func removeResource(_ *zap.Logger, ct *contract.Contract, c *kafkainternals.Consumer) int {
+	idx := coreconfig.FindResource(ct, c.GetUID())
+	if idx == coreconfig.NoResource {
+		return coreconfig.NoResource
+	}
+	coreconfig.DeleteResource(ct, idx)
+	return idx
+}
+
+func (r Reconciler) schedule(ctx context.Context, logger *zap.Logger, c *kafkainternals.Consumer, mutatorFunc contractMutatorFunc) (bool, error) {
+	// Get the data plane pod when the Consumer should be scheduled.
+	p, err := r.getPod(c)
+	if apierrors.IsNotFound(err) {
+		// Pod not found, return no error since the Consumer
+		// will get re-queued when the pod is added.
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	// Check if the pod is running.
+	if p.Status.Phase != corev1.PodRunning {
+		return false, nil
+	}
+
+	// Get contract associated with the pod.
+	b, cm, ct, err := r.getContract(ctx, c, p, logger)
+	if err != nil {
+		return false, err
+	}
+
+	if changed := mutatorFunc(logger, ct, c); changed == coreconfig.ResourceChanged {
+		logger.Debug("Contract changed", zap.Int("changed", changed))
+
+		ct.IncrementGeneration()
+
+		if err := b.UpdateDataPlaneConfigMap(ctx, ct, cm); err != nil {
+			return false, err
+		}
+	}
+
+	return true, b.UpdatePodsAnnotation(ctx, logger, "dispatcher" /* component, for logging */, ct.Generation, []*corev1.Pod{p})
+}
+
+func (r Reconciler) getPod(c *kafkainternals.Consumer) (*corev1.Pod, error) {
+	p, err := r.PodLister.Pods(c.Spec.PodBind.PodNamespace).Get(c.Spec.PodBind.PodName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pod %s/%s: %w", c.Spec.PodBind.PodNamespace, c.Spec.PodBind.PodName, err)
+	}
+	return p, nil
+}
+
+func (r Reconciler) getContract(ctx context.Context, c *kafkainternals.Consumer, p *corev1.Pod, logger *zap.Logger) (base.Reconciler, *corev1.ConfigMap, *contract.Contract, error) {
```

Comment:
should we name them differently to reconcile, instead of `get*` ?

Reply:
I'll rename them.